stdlib/lists.ncl

```diff
@@ -1,21 +1,31 @@
 {
   lists = {
+    head : List -> Dyn = fun l => %head% l;
+
+    tail : List -> List = fun l => %tail% l;
+
+    length : List -> Num = fun l => %length% l;
+
+    map : (Dyn -> Dyn) -> List -> List = fun f l => %map% f l;
+
+    elemAt : List -> Num -> Dyn = fun l n => %elemAt% l n;
+
     concat : List -> List -> List = fun l1 l2 => l1 @ l2;
 
     foldl : forall a. (a -> Dyn -> a) -> a -> List -> a =
       fun f fst l =>
-        if length l == 0 then
+        if %length% l == 0 then
```

Comment:
Unrelated to this PR, but I hope that `%length% l` is O(1). Otherwise `foldl` (and `fold`) as written here is quadratic.

Reply:
Yes, list are currently implemented as vectors so `length` is O(1). I suspect this was written because at that time equality was implemented only for numbers though, so `l == []` is probably more future-proof.